src/index/bip352.cpp

```diff
@@ -74,6 +74,14 @@ bool BIP352Index::GetSilentPaymentKeys(std::vector<CTransactionRef> txs, CBlockU
 
         std::optional<CPubKey> tweaked_pk = BIP352::GetSerializedSilentPaymentsPublicData(tx->vin, coins);
         if (tweaked_pk) {
+            // Used to filter dust. To keep the index small we use only one byte
+            // and measure in hexasats.
+            uint8_t max_output_amount = UINT8_MAX; // 4080 sat
+            for (const CTxOut& out : tx->vout) {
+                // TODO: only consider Taproot outputs
+                max_output_amount = std::min(out.nValue, CAmount(max_output_amount) << dust_shift) >> dust_shift;
```

Comment:
835dd4ba1ca4cc1abfd338e6bc38e25a116f5f8b: oh this is wrong, it finds the _lowest_ output amount, oops.